src/frontends/paddle/tests/test_models/gen_scripts/save_model.py

```diff
@@ -3,9 +3,225 @@
 
 import os
 import sys
+import errno
 import numpy as np
 import paddle
 
+if paddle.__version__ < "2.6.0":
+    from paddle.fluid.framework import (
+        Program,
+        Parameter,
+        default_main_program,
+        default_startup_program,
+        Variable,
+        program_guard,
+        dygraph_not_support,
+        static_only,
+    )
+    from paddle.fluid import core
+else:
+    from paddle.base.framework import (
+        Program,
+        Parameter,
+        default_main_program,
+        default_startup_program,
+        Variable,
+        program_guard,
+        dygraph_not_support,
+        static_only,
+    )
+    from paddle.base import core
+
+
+def prepend_feed_ops(
+    inference_program, feed_target_names, feed_holder_name='feed'
+):
+    if len(feed_target_names) == 0:
+        return
+
+    global_block = inference_program.global_block()
+    feed_var = global_block.create_var(
+        name=feed_holder_name,
+        type=core.VarDesc.VarType.FEED_MINIBATCH,
+        persistable=True,
+    )
+
+    for i, name in enumerate(feed_target_names):
+        if not global_block.has_var(name):
+            raise ValueError(
+                "The feeded_var_names[{i}]: '{name}' doesn't exist in pruned inference program. "
+                "Please check whether '{name}' is a valid feed_var name, or remove it from feeded_var_names "
+                "if '{name}' is not involved in the target_vars calculation.".format(
+                    i=i, name=name
+                )
+            )
+        out = global_block.var(name)
+        global_block._prepend_op(
+            type='feed',
+            inputs={'X': [feed_var]},
+            outputs={'Out': [out]},
+            attrs={'col': i},
+        )
+
+
+def append_fetch_ops(
+    inference_program, fetch_target_names, fetch_holder_name='fetch'
+):
+    global_block = inference_program.global_block()
+    fetch_var = global_block.create_var(
+        name=fetch_holder_name,
+        type=core.VarDesc.VarType.FETCH_LIST,
+        persistable=True,
+    )
+
+    for i, name in enumerate(fetch_target_names):
+        global_block.append_op(
+            type='fetch',
+            inputs={'X': [name]},
+            outputs={'Out': [fetch_var]},
+            attrs={'col': i},
+        )
+
+@static_only
+def save_inference_model(
```

Comment:
There is 'save_inference_model' in paddle 2.6.0. Why implement it again here?

Reply:
Why does it require to modify almost all test files? paddle.static.save_inference_model is the same as paddle.static.io.save_inference_model. We have already used this API in save_model.py

Reply:
> Why does it require to modify almost all test files? paddle.static.save_inference_model is the same as paddle.static.io.save_inference_model. We have already used this API in save_model.py

You are right. `paddle.static.save_inference_model` is the same as `paddle.static.io.save_inference_model`.

However, for most of the test files use the `saveModel` with `use_static_api=False`. We must to change chose test files because those files will have the following error when directly using `paddle.static.save_inference_model`.

<img width="1508" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/55493212/b8e764d9-a34f-42eb-9c7e-68c091a7d9b0">

for example:

```
# src/frontends/paddle/tests/test_models/gen_scripts/generate_deformable_conv.py

saveModel(name, exe, feedkeys=list(feed_dict.keys()), fetchlist=[node_out],
                  inputs=inputs_list, outputs=outs, target_dir=sys.argv[1] if len(sys.argv) > 1 else '.')

# change to

feed_list = [node_x, node_offset, node_weight]
if mask is not None:
    feed_list.append(node_mask)
if bias is not None:
    feed_list.append(node_bias)
saveModel(name, exe, feedkeys=feed_list, fetchlist=[node_out],
          inputs=inputs_list, outputs=outs, target_dir=sys.argv[1] if len(sys.argv) > 1 else '.', use_static_api=True)

```

Reply:
It may make others confused if implement saveModel in 'save_model.py'. I prefer to change all the feed_list like the example above. I also advise to remove 'use_static_api' parameter and choose the correct save_inference_model API according to the paddle version in saveModel. Can you help to do this?

Reply:
> It may make others confused if implement saveModel in 'save_model.py'. I prefer to change all the feed_list like the example above. I also advise to remove 'use_static_api' parameter and choose the correct save_inference_model API according to the paddle version in saveModel. Can you help to do this?

ok, I'll help with it. But it seems that paddle.static.io.save_inference_model is suitable for paddle >= 2.0.0. We do not need judge paddle version in saveModel. How about use paddle.static.io.save_inference_model for all causes?

Reply:
Yes. I think it's ok.
It is a lot of work. Very appreciate to help.